rllib/agents/ppo/ppo_tf_policy.py

```diff
@@ -110,9 +115,12 @@ def reduce_mean_valid(t):
         vf_loss = mean_vf_loss = tf.constant(0.0)
 
     total_loss = reduce_mean_valid(-surrogate_loss +
-                                   policy.kl_coeff * action_kl +
                                    policy.config["vf_loss_coeff"] * vf_loss -
                                    policy.entropy_coeff * curr_entropy)
+    # Add mean_kl_loss (already processed through `reduce_mean_valid`),
+    # if necessary.
+    if policy.config["kl_coeff"] > 0.0:
```

Comment:
seems this if statement is not necessary, 0.0 case, mean_kl_loss is zero. same below.

Reply:
Normally, yes, but if the calculated kl is Inf (due to some actions having almost 0.0 probability -> log(a) = -inf), then multiplying it with 0.0 will result in NaN. :)